server/render/index.js

```diff
@@ -86,8 +86,7 @@ export function serverRender( req, res ) {
 		config.isEnabled( 'server-side-rendering' ) &&
 		context.layout &&
 		! context.user &&
-		isEmpty( context.query ) &&
```

Comment:
I don't think this change implies calling `renderToString` more often. Here we're basically saying that we should ignore query parameters and just cache the first version of the page rendered. Follow up requests with the same pathname but different query parameters will end up loading that first version.
I agree it's quite a change but I tested the few logged out routes we have and I didn't see any issue with it so far.